dolphinscheduler-worker/src/main/docker/Dockerfile

```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 
-FROM eclipse-temurin:8-jre
+FROM eclipse-temurin:8-jdk
```

Comment:
It's better to replace all `Dockerfile` to `jre`.

Reply:
I think it's better to keep things the way they are. Because only `Worker Server` and `Standalone Server` require JDK, JRE is sufficient for other components. Unless other components need it as well.

Reply:
You are right. So only `Standalone` needs to be added.

Reply:
If the task that relies on JAVA_HOME uses an environment to import the JAVA_HOME, do we still need to change the Docker file?

Reply:
> If the task that relies on JAVA_HOME uses an environment to import the JAVA_HOME, do we still need to change the Docker file?

@ruanwenjun 
Yes, because the JAVA task requires `javac` to compile java code, and this command requires the JDK to be installed, which is not available when only the JRE is installed.

Reply:
> If the task that relies on JAVA_HOME uses an environment to import the JAVA_HOME, do we still need to change the Docker file?

@ruanwenjun 
If the `worker server` does not provide a basic working environment, users will have to install jdk and rebuild the docker image themselves, which can avoid some security issues, but it is not very user friendly.

Reply:
Please add this change to `Standalone` together.

Reply:
> Please add this change to `Standalone` together.

Standalone Server already uses a docker image with JDK